librz/bin/p/bin_nes.c

```diff
@@ -101,13 +101,13 @@ static RzList *sections(RzBinFile *bf) {
 	ptr->name = strdup("ROM");
 	ptr->paddr = INES_HDR_SIZE;
 	ptr->size = ihdr.prg_page_count_16k * PRG_PAGE_SIZE;
+	bool mirror = ROM_START_ADDRESS + ptr->size <= ROM_MIRROR_ADDRESS; // not a 256bit ROM, mapper 0 mirrors the complete ROM in this case
 	ptr->vaddr = ROM_START_ADDRESS;
-	ptr->vsize = ROM_SIZE;
+	ptr->vsize = mirror ? ROM_MIRROR_ADDRESS - ROM_START_ADDRESS : ROM_SIZE; // make sure the ROM zero excess does not overlap the mirror
```

Comment:
Minor change in behavior here, for the change in `librz/core/cbin.c`
Before, the bin plugin would suggest a mapping like this:
```
 4 fd: 3 +0x00000010 0x00008000 - 0x0000bfff r-x fmap.ROM
 3 fd: 4 +0x00000000 0x0000c000 - 0x0000ffff r-x mmap.ROM
 2 fd: 3 +0x00000010 0x0000c000 - 0x0000ffff r-x fmap.ROM_MIRROR
 1 fd: 4 +0x00000000 0x00010000 - 0x00013fff r-x mmap.ROM_MIRROR
 ```
Before the `cbin.c` change, core would not map the `mmap.ROM` one, but now it does and by default it overlaps the `fmap.ROM_MIRROR` so the data there would be falsely zero. Now I made the bin plugin to not let the maps overlap.
An alternative would be to swap the priority but I think it makes more sense this way.